### DISCUSSION TITLE
Sketch tft_setup.h not working

### DISCUSSION THREAD

**Discussion by griffinmt** [Bodmer/TFT_eSPI#2352] [Compatible displays and setup files]
I read in TFT_eSPI.h that you can put your specific setup file with your sketch and naming it tft_setup.h. I am using Arduino IDE 1.8.19 and TFT_eSPI 2.5
It does not seem to work for me. What am I probably doing wrong again :)

**Comment by GeorgeIoak** [comment]
Take a look here, https://github.com/Bodmer/TFT_eSPI#tips

**Comment by Bodmer** [comment]
Adding the setup file to the sketch folder is no longer recommended as there are undesirable consequences in some scenarios.

See #2007

**Reply by mcbp223** [reply]
Hi, I use this method too, it's very useful, did you see any issues other than the cache problem on IDE 2.0?

On my machine I've set up `platform.local.txt` with
`build.extra_flags="-I{build.source.path}/libinc" "-I{build.source.path}/../libinc" -I/home/me/Arduino/libinc`
so I put `tft_setup.h`, also `lv_conf.h` in these subdirectories named libinc instead of the sketch folder directly, to avoid potential name clashes with sketch files. Then in `tft_setup.h` I indicate where the file came from, with 
```cpp
#warning "libdir/tft_setup.h found in sketchbook"
```
and another message if it was included from within the sketch or other folder. Having one in the sketchbook allows compiling examples without changing them,  and one at a level outside the sketch allows for some workbench scenarios.

I've seen efforts at Arduino to make the [custom boards method](https://arduino.github.io/arduino-cli/0.29/platform-specification/#referencing-another-core-variant-or-tool) work for other vendor platforms too, that would be much better, but a bit complicated for beginners. Until then, it's only fair to use the `platform.local.txt` method instead of modifying the library all the time.

**Comment by griffinmt** [comment]
I have several different displays. Is it possible to make several setup
files and pass which one to use from my sketch?

**Comment by GeorgeIoak** [comment]
If you switch to using PlatforIO IDE extension with VS Code then this is very easy to do since all of your display configuration is added to the platformio.ini file as build flags.
